src/simulators/matrix_product_state/matrix_product_state.hpp

```diff
@@ -375,6 +392,17 @@ void State::set_config(const json_t &config) {
   }
 }
 
+void State::add_metadata(ExperimentData &data, const json_t &config) const {
+  double threshold;
+  if (JSON::get_value(threshold, "matrix_product_state_truncation_threshold", config))
+    data.add_metadata("matrix_product_state_truncation_threshold", threshold);
```

Comment:
These values could be stored as class variables, so you don't need to get them again from the json